### PR TITLE
PLAT-4390 iOS - Provide data for sync improvements

### PR DESCRIPTION
**Pull request by alex-moc**
send notification into to two hypermedia links

### CODE REVIEW

File: FitpaySDK/Notifications/NotificationDetail.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    open var deviceId: String?
+    open var userId: String?
+    open var clientId: String?
+    var restClient:RestClient?
+
+    public required init?(map: Map) {
+    }
+    
+    open func mapping(map: Map) {
+        ackSync <- map["_links.ackSync.href"]
```
Comment (ssteveli) [inline FitpaySDK/Notifications/NotificationDetail.swift]: Just confirming that `_links` and `_links.ackSync` and `_links.completeSync` are all optional and this won't throw any type of error if any of them don't exist.   Correct?

File: FitpaySDK/Notifications/NotificationDetail.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        guard let client = self.restClient else {
+            log.error("SYNC_ACKNOWLEDGMENT: trying to send ackSync without rest client.")
+            return
+        }
+
+        client.makePostCall(ackSync, parameters:nil) { (error) in
+            if (error != nil) {
+                log.error("SYNC_ACKNOWLEDGMENT: ackSync failed to send.")
```
Comment (ssteveli) [inline FitpaySDK/Notifications/NotificationDetail.swift]: Should the `error` be logged or is it already logged elsewhere?

File: FitpaySDK/Rest/Models/CommitMetrics.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        
+        guard let client = self.notificationAsc?.restClient else {
+            log.error("SYNC_ACKNOWLEDGMENT: trying to send completeSync without rest client.")
+            return
+        }
+        
+        let params = ["params": self.toJSON()]
+        client.makePostCall(completeSync, parameters: params as [String: AnyObject]?) { (error) in
+            if (error != nil) {
+                log.error("SYNC_ACKNOWLEDGMENT: completeSync failed to send.")
```
Comment (ssteveli) [inline FitpaySDK/Rest/Models/CommitMetrics.swift]: Should `error` be logged or is it logged elsewhere?

File: FitpaySDK/Notifications/NotificationDetail.swift (hunk 1 above)
Comment (alex-moc) [reply FitpaySDK/Notifications/NotificationDetail.swift]: All fields are optional

**Comment by alex-moc** [comment]
@Clark73 
I can add the additional optional parameter to
SyncRequestQueue.sharedInstance.add(request:, completion:)
like this:
SyncRequestQueue.sharedInstance.add(request:, payload, completion:)

**Comment by Clark73** [comment]
That would work!

**Pull request merged**